consensus/qsc/mod.go

```diff
@@ -67,38 +69,59 @@ func (c *Consensus) GetChain(id []byte) (consensus.Chain, error) {
 // primitives to send proposals to a network of nodes.
 func (c *Consensus) Listen(val consensus.Validator) (consensus.Actor, error) {
 	go func() {
-		// TODO: how to stop the loop ?
 		for {
-			err := c.executeRound(val)
+			var proposal consensus.Proposal
+			select {
+			case <-c.closing:
+				fabric.Logger.Trace().Msg("closing")
+				return
+			case proposal = <-c.ch:
+			default:
+				// If the current node does not have anything to propose, it
+				// still has to participate so it sends an empty proposal.
+				proposal = nil
+			}
+
+			ctx, cancel := context.WithTimeout(context.Background(), EpochTimeout)
+
+			go func() {
+				// This Go routine is responsible for listening a close event
+				// from the actor.
+				select {
+				case <-ctx.Done():
```

Comment:
no check of `ctx.Error()`?

Reply:
If there is an error, it will be propagated by the `executeRound` function and then logged.